src/data/directory_content.rs

```diff
@@ -114,6 +168,19 @@ impl DirectoryEntry {
     }
 }
 
+#[cfg(feature = "metadata_view")]
+impl DirectoryEntry {
+    /// Clones the content of the directory item, if available
+    pub fn content(&self) -> Option<String> {
+        self.content.clone()
+    }
+
+    /// Sets the content of the directory item
+    pub fn set_content(&mut self, content: Option<String>) {
+        self.content = content;
+    }
```

Comment:
For consistency with the other public methods, this should return a mutable reference:
```rust
pub fn content_mut(&mut self) -> Option<&mut String> {
    ....
}
```

Reply:
```rust
        if let Some(item) = &mut self.selected_item {
            *item.content_mut() = Some(item);
        }
```
I was just thinking about it, we can leave `set_content` as well. It's easier to use.